src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/ShutdownTask.java

```diff
@@ -107,11 +104,13 @@ public TaskResult call() {
              * workers to contend for the lease of this shard.
              */
             if(localReason == ShutdownReason.TERMINATE) {
-                latestShards = kinesisProxy.getShardList();
+                ShardClosureVerificationResponse shardClosureVerificationResponse = kinesisProxy.verifyShardClosure(shardInfo.getShardId());
+                latestShards = shardClosureVerificationResponse.getLatestShards();
 
-                //If latestShards is null or empty, we should still shut down the ShardConsumer with Zombie state which avoid
-                // checking point with SHARD_END sequence number.
-                if(CollectionUtils.isNullOrEmpty(latestShards) || !isShardInContextParentOfAny(latestShards)) {
+                // If latestShards is null or empty, it means shard in context is not closed yet.
```

Comment:
I latestShards is null or empty, we do not have enough info to take a decision. Hence we go ahead and shut downt he consumer